### PR TITLE
Codegen improvements

### PR DESCRIPTION
**Pull request by janosbrodbeck**
* Generated code of robots does not have the ID anymore in the filename. The ID is always hidden for the user, it has no meaning for them in the filename
* Make `x_rel`, `y_rel`, `z_rel` available as additional expressions in templates. It is the relative position of the welding point to the robot

### CODE REVIEW

File: src/app/codegen/code_generator.py [hunk 1]
```diff
@@ -49,13 +53,25 @@ def generate(template: GenerationTemplate, welding_points: List[WeldingPoint]) -
         # Define a dictionary with functions, which can be used in templates
         func_dict = {"generate_blockly_block_id": TemplateFunctions.generate_blockly_block_id}
 
+        # Robot for which the program will be generated
+        robot = welding_points[0].robot
+        if robot is None:
+            raise CodeGeneratorException("Welding point does not have a robot")
```
Comment (dolevi-tub) [inline src/app/codegen/code_generator.py:59]:
```suggestion
        robots = set([wp.robot.id for wp in welding_points])
        if len(robots) !== 1 or robots[0] is None:
            raise CodeGeneratorException("All welding points must have the same non-None robot")
        robot = welding_points[0].robot
```
I think we should rather validate whether all the robots have the same non-None robot. (Although "non-None" does sound awful)

**Review by dolevi-tub — APPROVED** [review]
I know the recommendation doesn't improve our application effectively as `CodeGenerator.generate()` is ensured to be called with only one robot type. So you can merge too

File: src/app/codegen/code_generator.py (hunk 1 above)
Comment (janosbrodbeck) [reply src/app/codegen/code_generator.py]: From application perspective, it's also not possible that this method gets called with a `None` robot type (except for the unit test), but stuff like this should still be checked. Thanks for the catch!

File: src/app/codegen/code_generator.py [hunk 2]
```diff
@@ -54,9 +54,10 @@ def generate(template: GenerationTemplate, welding_points: List[WeldingPoint]) -
         func_dict = {"generate_blockly_block_id": TemplateFunctions.generate_blockly_block_id}
 
         # Robot for which the program will be generated
+        robots = set([wp.robot.id for wp in welding_points])
+        if len(robots) !== 1 or robots[0] is None:
```
Comment (janosbrodbeck) [inline src/app/codegen/code_generator.py:58]:
```suggestion
        if len(robots) != 1 or robots[0] is None:
```

**Review by dolevi-tub — APPROVED** [review]

**Comment by janosbrodbeck** [comment]
Thanks for fixing :)

**Pull request merged**